buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersPlugin.java

```diff
@@ -140,33 +140,82 @@ private void createListClustersTask(Project project, NamedDomainObjectContainer<
+    private void createUseClusterTaskExtensions(Project project, NamedDomainObjectContainer<ElasticsearchCluster> container) {
         // register an extension for all current and future tasks, so that any task can declare that it wants to use a
         // specific cluster.
         project.getTasks().all((Task task) ->
             task.getExtensions().findByType(ExtraPropertiesExtension.class)
-                .set(
-                    "useCluster",
-                    new Closure<Void>(project, task) {
-                        public void doCall(ElasticsearchCluster cluster) {
-                            if (container.contains(cluster) == false) {
-                                throw new TestClustersException(
-                                    "Task " + task.getPath() + " can't use test cluster from" +
+            .set(
+                "useCluster",
+                new Closure<Void>(project, task) {
+                    public void doCall(ElasticsearchCluster cluster) {
+                        Object thisObject = this.getThisObject();
+                        if (thisObject instanceof Task == false) {
+                            throw new AssertionError("Expected " + thisObject + " to be an instance of " +
+                                "Task, but got: " + thisObject.getClass());
+                        }
+                        if (container.contains(cluster) == false) {
+                            throw new TestClustersException(
+                                "Task " + task.getPath() + " can't use test cluster from" +
                                     " another project " + cluster
-                                );
-                            }
-                            Object thisObject = this.getThisObject();
-                            if (thisObject instanceof Task == false) {
-                                throw new AssertionError("Expected " + thisObject + " to be an instance of " +
-                                    "Task, but got: " + thisObject.getClass());
-                            }
-                            usedClusters.computeIfAbsent(task, k -> new ArrayList<>()).add(cluster);
-                            ((Task) thisObject).dependsOn(
-                                project.getRootProject().getTasks().getByName(SYNC_ARTIFACTS_TASK_NAME)
                             );
                         }
-                    })
+                        usedClusters.computeIfAbsent(task, k -> new ArrayList<>()).add(cluster);
+                        task.dependsOn(
+                            project.getRootProject().getTasks().getByName(SYNC_ARTIFACTS_TASK_NAME)
+                        );
+                    }
+                })
         );
+
+        project.getTasks().all(task ->
+            task.getExtensions().findByType(ExtraPropertiesExtension.class)
+            .set(
+                "chainCluster",
```

Comment:
This name is a bit confusing. Especially since it implies you should pass a cluster, but in fact it expects a `Task`. I'm struggling to find a better name though.